### PR TITLE
[Merged by Bors] - Additional networking metrics

### PR DESCRIPTION
**Pull request by AgeManning**
Adds additional metrics for network monitoring and evaluation.

### CODE REVIEW

**Comment by AgeManning** [comment]
This will resolve #2553

**Comment by AgeManning** [comment]
There's a bug in libp2p that needs to be resolved before we can have nice things

**Comment by ethDreamer** [comment]
maybe i'm confused - but i *thought* the flag to enable gossipsub metrics was `metrics` and not `gossipsub-metrics`?

**Comment by AgeManning** [comment]
I'm in the process of removing the feature flag. But, to answer your question, the `metrics` flag is for the `gossipsub` crate, but we are importing `libp2p` and the `gossipsub-metrics` flag enables `metrics` on gossipsub.

**Comment by AgeManning** [comment]
@divagant-martian - This is almost ready, once we merge down the libp2p deps. Could you give this guy a review in anticipation of the libp2p PR merge

File: beacon_node/http_metrics/src/lib.rs
Comment (divagant-martian) [inline beacon_node/http_metrics/src/lib.rs:43]: why `std`'s `Mutex` over `parking_lot`'s?

File: beacon_node/lighthouse_network/src/peer_manager/mod.rs
Comment (divagant-martian) [inline beacon_node/lighthouse_network/src/peer_manager/mod.rs:879]: I think you'd want to check if metrics are enabled here too

File: beacon_node/lighthouse_network/src/peer_manager/peerdb/client.rs
Comment (divagant-martian) [inline beacon_node/lighthouse_network/src/peer_manager/peerdb/client.rs]: can be removed

File: beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs
Comment (divagant-martian) [inline beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs:392]: yeah finally

**Review by divagant-martian — COMMENTED** [review]
Minor comments. I like the changes

File: beacon_node/http_metrics/src/lib.rs
Comment (AgeManning) [reply beacon_node/http_metrics/src/lib.rs:43]: I think the mutex only holds between multiple calls the to the HTTP_API. 

Wasn't sure about the mutability in reading the registry, so just used a std mutex. If we only read it in each HTTP_API call, probably better to use parking lot and take a read lock (if that is sufficent). If we are taking a write lock every time, should be the same as a usual mutex right?

Comment (AgeManning) [reply beacon_node/http_metrics/src/lib.rs:43]: Lets revisit this and update to parking lot if we want

**Comment by AgeManning** [comment]
This PR removes the naming for the `libp2p_connected_peers` metric and uses the more standardized `libp2p_peers`. This means some dashboards may not function as expected after this merge

File: beacon_node/http_metrics/src/lib.rs
Comment (michaelsproul) [reply beacon_node/http_metrics/src/lib.rs:43]: Parking lot has a `Mutex` that's a drop in replacement for `std`'s, except that it's faster and errors in fewer scenarios:

https://docs.rs/parking_lot/0.11.2/parking_lot/type.Mutex.html

The `std` Mutex is also fine though, so I think we can merge as-is

File: beacon_node/network/src/service.rs
Comment (michaelsproul) [inline beacon_node/network/src/service.rs]:
```suggestion
```

**Review by michaelsproul — APPROVED** [review]
LGTM modulo one out of date comment!

I also [left a comment about `parking_lot::Mutex`](https://github.com/sigp/lighthouse/pull/2549#discussion_r773554296), but we can leave it as-is

**Comment by michaelsproul** [comment]
Looks like there are a few tests in need of updating too

**Pull request closed without merging**
